MacOS CI broken

### Prerequisites

na

### Description

The tests for macOS do not, with a missing modules error instead. It seems node modules are not being loaded, or something to that effect.

### Steps to Reproduce

1. See [the result](https://github.visualstudio.com/Atom/_build/results?buildId=43244) of a PR that only changes doc comments.

**Expected behavior:**

MacOS tests run

**Actual behavior:**
```
[command]/bin/bash --noprofile --norc /Users/vsts/agent/2.153.1/work/_temp/5b8919b0-5d5f-4616-a604-fb5e95a6970c.sh
internal/modules/cjs/loader.js:596
    throw err;
    ^

Error: Cannot find module 'colors'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Module.require (internal/modules/cjs/loader.js:650:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/vsts/agent/2.153.1/work/1/s/script/test:5:1)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

**Reproduces how often:**

100%. This happened on [another PR](https://github.com/atom/atom/pull/18738) too.

### Versions

na

--- a/script/vsts/platforms/macos.yml
+++ b/script/vsts/platforms/macos.yml
@@ -122,6 +122,16 @@ jobs:
           targetfolder: '**/node_modules, !**/node_modules/**/node_modules'
           vstsFeed: 'bae1bc26-220d-43c7-a955-4de039370de2'
 
+      # The artifact caching task does not work on forks, so we need to
+      # bootstrap again for pull requests coming from forked repositories.
+      - script: script/bootstrap
+        displayName: Bootstrap build environment
+        env:
+          CI: true
+          CI_PROVIDER: VSTS
+          NPM_BIN_PATH: /usr/local/bin/npm
+        condition: ne(variables['CacheRestored'], 'true')
+
       - task: DownloadBuildArtifacts@0
         displayName: Download atom-mac.zip
         inputs:

